### PR TITLE
some tests around workflow mocking

### PR DESCRIPTION
**Pull request by bergman**
# TL;DR
Some additional test cases for #218 to document current behaviour

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit-testing/src/test/java/org/flyte/flytekit/testing/SdkTestingExecutorTest.java [hunk 1]
```diff
@@ -199,6 +199,148 @@ public TestUnaryStringIO expand(SdkWorkflowBuilder builder, TestUnaryStringIO in
         equalTo("Fixed input [string] (of type INTEGER) doesn't match expected type STRING"));
   }
 
+  @Test
+  public void testWithTask_mockedRunnableTasksAreNotRun() {
```
Comment (honnix) [inline flytekit-testing/src/test/java/org/flyte/flytekit/testing/SdkTestingExecutorTest.java:203]: I think this is covered by existing tests.

File: flytekit-testing/src/test/java/org/flyte/flytekit/testing/SdkTestingExecutorTest.java [hunk 2]
```diff
@@ -199,6 +199,148 @@ public TestUnaryStringIO expand(SdkWorkflowBuilder builder, TestUnaryStringIO in
...
+    SdkWorkflow<Void, SumTask.SumOutput> workflow =
+        new SdkWorkflow<>(SdkTypes.nulls(), JacksonSdkType.of(SumTask.SumOutput.class)) {
+          @Override
+          public SumTask.SumOutput expand(SdkWorkflowBuilder builder, Void noInput) {
+            builder.apply(
+                new SumTask(),
+                SumTask.SumInput.create(
+                    SdkBindingDataFactory.of(1L),
+                    SdkBindingDataFactory.of(1L)));
+            SdkBindingData<Long> mockedSum2 = builder.apply(
```
Comment (honnix) [inline flytekit-testing/src/test/java/org/flyte/flytekit/testing/SdkTestingExecutorTest.java:243]:
```suggestion
            SdkBindingData<Long> sum2 = builder.apply(
```

File: flytekit-testing/src/test/java/org/flyte/flytekit/testing/SdkTestingExecutorTest.java [hunk 3]
```diff
@@ -199,6 +199,148 @@ public TestUnaryStringIO expand(SdkWorkflowBuilder builder, TestUnaryStringIO in
...
+                new SumTask(),
+                SumTask.SumInput.create(
+                    SdkBindingDataFactory.of(1L),
+                    SdkBindingDataFactory.of(1L)))
+                .getOutputs()
+                .c();
+            return SumTask.SumOutput.create(mockedSum2);
```
Comment (honnix) [inline flytekit-testing/src/test/java/org/flyte/flytekit/testing/SdkTestingExecutorTest.java:250]:
```suggestion
            return SumTask.SumOutput.create(sum2);
```

File: flytekit-testing/src/test/java/org/flyte/flytekit/testing/SdkTestingExecutorTest.java (hunk 1 above)
Comment (bergman) [reply flytekit-testing/src/test/java/org/flyte/flytekit/testing/SdkTestingExecutorTest.java:203]: I thought so too, but it looks like the other tests only tested remote tasks and I wanted an explicit test for local tasks. I know the test executor treats them no different but I was thinking it's good to be explicit (and I'm not sure of a better way to specify these requirements than a test).

**Comment by bergman** [comment]
Moved to #222
